apps/dashr-ncbi-explorer/app.R

```diff
@@ -0,0 +1,654 @@
+          htmlButton(
+            id = 'reset-file', n_clicks = 0, children = 'Clear FASTA File'
+          ),
+          htmlA(
+            htmlButton(
+              "Download FASTA data",
+              className="control-download"
+            ),
+            href="data/random_fasta.FASTA",
+            download="random_fasta.FASTA"
+          ),
+          
+          htmlDiv(id='alignment-file-upload-container', children=list(
+            dccUpload(
+              id='alignment-file-upload',
+              className='control-upload',
+              children=htmlDiv(list(
+                "Drag and drop FASTA files or select files."
+              ))
+            )
+          )),
+          
+          dccStore(
+            id = 'genbank-sequence'),
+          
+          dccStore(
+            id = 'fasta-file'
+          ),
+          
+          dccStore(
+            id = 'dataframe-store'
+          )
+        ))
+      ))
+    )
+  )), className = 'item-a'),
+  
+  htmlDiv(list(
+    htmlP("Multiple Sequence Alignment Chart", 
+          style = list("font-family" = "Open Sans", "font-size" = "22px", "color" = "#262B3D", "text-align" = "center")),
+    dccLoading(type = "dot", children = htmlDiv(
+      id = 'alignment-container',
+      children = list(),
+      style = list("overflow-x" = "scroll", "display" = "flex", "justify-content" = "center")
+    ))), className = 'item-b'),
+  
+  htmlDiv(children = list(
+    htmlP("Sequence Viewer", style = list("font-family" = "Open Sans", "font-size" = "22px", "color" = "#262B3D", "text-align" = "center")),
+    dashbioSequenceViewer(
+      id = 'sequence-viewer',
+      sequence = as.character(FASTA_DATA$`sp|Q9W678|P53_BARBU Cellular tumor antigen p53 OS=Barbus barbus GN=tp53 PE=2 SV=1`),
+      selection = list(10,20, "green"),
+      charsPerLine = 40
+    )), className = 'item-d'),
+  
+  htmlDiv(list(
+    dccGraph(id = "gc_graph")
+  ), className = 'item-e'),
+  
+  htmlDiv(dccGraph(
+    id = 'sequence-pie-chart'
+  ), className = 'item-f'),
+  
+  htmlDiv(children = list(
+    htmlP("Selected Dataset", style = list("font-family" = "Open Sans", "font-size" = "14px", "color" = "#262B3D", "text-align" = "center")),
+    dccRadioItems(
+      id = "cpg-options",
+      options = list(list(label = "Default", value = 1)),
+      value = 1),
+    htmlButton(
+      id = 'reset-button', n_clicks = 0, children = 'Refresh',
+      style = list("width" = "100%", "padding" = "0px", "margin" = "0px", "margin-top" = "15px")
+      )
+    )
+    ,className = 'item-g')
+  
+), className = "container")
+
+app$layout(
+  header,
+  first_row
+)
+
+# Callbacks for Sequence Viewer Output
+
+app$callback(
+  output(id = 'genbank-sequence', property = 'data'),
+  params = list(
+    input(id = 'submit-button', property = 'n_clicks'),
+    state(id = 'genbank-input', property = 'value')
+  ),
+  
+  update_data <- function(n_clicks, accession_id) {
+    accession_id <- gsub("\\s", "", accession_id)
```

Comment:
i think `stringi` is a faster alternative to `gsub` once you get into bigger data

Reply:
Noted, will keep in mind for future apps. In this case after testing both I saw comparable speeds and they're both practically instantaneous.